datastore/datastore.go

```diff
@@ -253,6 +254,32 @@ func NewDataStore(scope string, cfg *ScopeCfg) (DataStore, error) {
 	return newClient(scope, cfg.Client.Provider, cfg.Client.Address, cfg.Client.Config, cached)
 }
 
+// NewDataStoreFromConfig creates a new instance of LibKV data store starting from the datastore config data
+func NewDataStoreFromConfig(dsc discoverapi.DatastoreConfigData) (DataStore, error) {
+	if dsc.Config == nil {
+		return nil, fmt.Errorf("store configuration has nil store config")
```

Comment:
Why should this fail ?
Today the configs are used for TLS and heartbeat. But there can be clusters without these configs.
So, failing for a nil `dsc.Config` doesn't seem appropriate.